### ISSUE TITLE
Input Resolver Code Gen issue with type alias

### ISSUE THREAD

**Issue by mohammadcazi** [99designs/gqlgen#1868]
Looks like this pr caused an issue in code gen: https://github.com/99designs/gqlgen/pull/1404

The generated code expects a resolver for inputs now but it fails to implement the resolver. Also, the model that was already matching using @gomodel directive is no longer matching:

```graphql
type Bar struct {
   a, b string
}

type FooType []Bar
type Baz struct {
    Foo FooType
}


input SomeInput @goModel(model: "Baz") {
   foo: [Bar!]! # this used to get mapped to the go struct, but now it doesn't. 
}

input Bar @goModel(model: "Bar") {
   a: String!
   b: String!
}
```

**Comment by frederikhors** [comment]
Please create a Github repository to reproduce the issue.

**Comment by mojtabacazi** [comment]
@frederikhors here is a sample repository for the issue. 

https://github.com/mojtabacazi/gql-issue-type-alias

Generated code with 0.14 works fine but not 0.15 nor 0.16. Here is the error from the project:

```
./main.go:13:23: cannot use &rsv (type *resolvers.Resolver) as type generated.ResolverRoot in field value:
        *resolvers.Resolver does not implement generated.ResolverRoot (missing SampleInput method)
```

**Comment by mojtabacazi** [comment]
@frederikhors Any updates?

**Comment by StevenACoffman** [comment]
I believe this is resolved with #1950
